src/ui/Analytics/PendingSearchAsYouTypeSearchEvent.ts

```diff
@@ -72,6 +73,10 @@ export class PendingSearchAsYouTypeSearchEvent extends PendingSearchEvent {
     this.templateSearchEvent.actionType = newCause.type;
   }
 
+  public modifyQueryContent(query: string) {
```

Comment:
What I don't like with that though, is that I have to call the private `queryContent` member something else, like `_queryContent`, because you cannot have two property with the same name, obviously.

So yeah, I prefer to stay with a method/function.